knowledge_repo/app/routes/posts.py

```diff
@@ -68,16 +70,18 @@ def render(path):
 
     comments = post.comments
     for comment in comments:
-        comment.author = db_session.query(User).filter(User.id == comment.user_id).first().username
+        author = db_session.query(User).filter(User.id == comment.user_id).first()
+        if author is not None:
```

Comment:
I think you're right, here... but the fact that I cannot tell you with 100% certainty off-the-cuff troubles me. The problem is that Python offers a whole heap of magic behaviours, depending on which methods are provided, and if anyone upstream of us decides that it would be a good idea to have SQL models (or other complex datatypes) be iterable, then doing a `not <object>` operation would lead to slower and possibly incorrect code. I prefer to be defensive against upstream changes by being more explicit in such comparisons, especially for more complex Python datatypes, so I don't have to worry about implementation details, or new weird bugs appearing in the future.

If you feel strongly about it though, I think it should be safe enough here. The crew managing sqlalchemy are probably to do anything crazy.